src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Orchestration.cs

```diff
@@ -72,11 +73,22 @@ public partial class ReindexDurableFunction
                     // Create a new orchestration with the same instance ID to process the remaining data
                     logger.LogInformation("Completed re-indexing the range {Range}. Continuing with new execution...", batchRange);
 
+                    WatermarkRange completed = input.Completed.Count == 0
```

Comment:
use `input.Completed == WatermarkRange.None` instead of `input.Completed.Count == 0`? I feel the former one more clearly tells the value has not been initialized yet.